modules/remotecontrol/index.js

```diff
@@ -64,136 +61,148 @@ class RemoteControl {
-     * is started.
+     * Handles remote control start messages.
+     *
+     * @param {number} id - the id of the request that will be used for the
+     * response.
+     * @param {string} sourceId - The source id of the desktop sharing stream.
      */
-    handlePermissionRequest(userInfo) {
-        this.windowManager.requestRemoteControlPermissions(userInfo)
-            .then(result => {
-                this.sendEvent({
-                    type: EVENT_TYPES.permissions,
-                    action: result ? PERMISSIONS_ACTIONS.grant
-                        : PERMISSIONS_ACTIONS.deny,
-                    userId: userInfo.userId
-                });
-                if(result) {
-                    this.start();
+    start(id, sourceId) {
+        const displays = electron.screen.getAllDisplays();
+
+        switch(displays.length) {
+            case 0:
+                this.display = undefined;
+            break;
+            case 1: // Linux
+                this.display = displays[0];
+            break;
+            // eslint-disable-next-line no-case-declarations
+            default: // > 1 display
+                const coordinates = sourceId2Coordinates(sourceId);
+                if(coordinates) { // Windows
+                    const { x, y } = coordinates;
+                    this.display = electron.screen.getDisplayNearestPoint({
+                        x: x + 1,
+                        y: y + 1
+                    });
+                } else { // ac OS
+                    // try to match displayID with sourceID
+                    const displayID = Number(sourceId.replace('screen:', ''));
+                    this.display
+                        = displays.find(display => display.id === displayID);
                 }
-            }).catch(e => {
-                console.error(e);
-            });
-    }
+        }
 
-    /**
-     * Starts processing the events.
-     */
-    start() {
-        this.started = true;
+        const response = {
+            id,
+            type: 'response'
+        };
+
+        if(this.display) {
+            this.started = true;
+            response.result = true;
+        } else {
+            response.error
+                = 'Error: Can\'t detect the display that is currently shared';
+        }
+
+        this.sendResponse(response);
     }
 
     /**
      * Stops processing the events.
      */
     stop() {
         this.started = false;
+        this.display = undefined;
     }
 
     /**
-     * Executes the passed event.
-     * @param {Object} event the remote-control-event.
+     * Executes the passed message.
+     * @param {Object} message the remote-control message.
      */
-    onRemoteControlEvent(event) {
-        if(!this.started && event.type !== EVENT_TYPES.permissions) {
+    onRemoteControlMessage(message) {
+        const { id, data } = message;
+
+        if(!this.started
+            && data.type != REQUESTS.start) {
```

Comment:
Could you explain to me why data.type has the start value? Do the messages keep sending that type? Does "start" mean "start doing this action"?